bin/reth/src/cli/mod.rs

```diff
@@ -245,6 +232,16 @@ pub enum Commands<C: ChainSpecParser, Ext: clap::Args + fmt::Debug> {
     Prune(prune::PruneCommand<C>),
 }
 
+impl<C: ChainSpecParser, Ext: clap::Args + fmt::Debug> Commands<C, Ext> {
+    fn chain_spec(&self) -> Option<&C::ChainSpec> {
+        match self {
+            Self::Node(cmd) => Some(&cmd.chain),
+            // For other subcommands that do not provide a chain, return None.
+            _ => None,
```

Comment:
we do have a few more instances where this is the case,

this will require a bit more copy paste work:

```
impl<C: ChainSpecParser> Command<C> {

    /// Returns the chainspec
    pub fn chain_spec(&self) -> &Arc<C::ChainSpec> {
        &self.env.chain
    }
}
```

basically on all commands the use a `<C>` generic

Reply:
Didn't know chain spec was buried inside `env`. Working on it, thanks!